projectroles/forms.py

```diff
@@ -631,21 +631,11 @@ def clean(self):
                     s_field = 'settings.{}.{}'.format(name, s_key)
 
                     if s_val['type'] == 'JSON':
-                        # for some reason, there is a distinct possibility, that the
-                        # initial value has been discarded and we get '' as value.
-                        # Seems to only happen in automated tests. Will catch that
-                        # here.
                         if not self.cleaned_data.get(s_field):
                             self.cleaned_data[s_field] = '{}'
-                        try:
-                            self.cleaned_data[s_field] = json.loads(
-                                self.cleaned_data.get(s_field)
-                            )
-                        except json.JSONDecodeError as err:
```

Comment:
I've investigated this bug and found out, that we need check for empty string in `self.cleaned_data[s_field]` in case of empty value in form. Check for valid JSON string is also needed as you mentioned above.
The only thing is that I changed `forms.ValidationError` to `self.add_error`. As a result there is no more a banner on the top of form, but a small hint near JSON field. Idk if I did it right, let me know.